forc-pkg/src/manifest.rs

```diff
@@ -61,6 +69,38 @@ impl ManifestFile {
             .parent()
             .expect("failed to retrieve manifest directory")
     }
```

Comment:
Let's also add a `pub fn lock_path(&self) -> Result<PathBuf>` method here so that it can be easily retrieved for passing to the `BuildPlan::from_lock_and_manifests` constructor.

This is a little tricky as, even if we have a `ManifestFile::Package`, we still want to check if our package is *within* a workspace. This is because if our package is within a workspace, we always want to use the workspace-level lock file to ensure a consistent set of pinned dependencies.

I'm thinking we could do something like the following:

1. Add a `pub fn workspace(&self) -> Result<Option<WorkspaceManifestFile>>` method to `PackageManifestFile` that returns a `WorkspaceManifestFile` in the case that the package is within a workspace, otherwise `None`.
2. Add a `pub fn lock_path(&self) -> PathBuf` method to the `WorkspaceManifestFile`.
3. Add a `pub fn lock_path(&self) -> Result<PathBuf>` method to `PackageManifestFile` that first checks for the workspace, falling back to a path for the package lock if no workspace.

Then our `ManifestFile::lock_path` function could look something like:

```rust
pub fn lock_path(&self) -> Result<PathBuf> {
    match self {
        Self::Package(ref pkg) => pkg.lock_path(),
        Self::Workspace(ref ws) => Ok(ws.lock_path()),
    }
}